pkg/istio-agent/xds_proxy.go

```diff
@@ -345,6 +347,13 @@ func (p *XdsProxy) handleStream(downstream adsStream) error {
 	return p.HandleUpstream(ctx, con, xds)
 }
 
+func (p *XdsProxy) buildUpstreamConn(ctx context.Context) (*grpc.ClientConn, error) {
+	p.optsMutex.Lock()
```

Comment:
nit: perhaps an RLock() ?

Reply:
can we fetch `p.istiodDialOptions` in lock, then call `grpc.DialContext` out of the lock? Otherwise we can block the lock for a long time

Reply:
Ping